docs/scripting-ref.rst

```diff
@@ -2102,3 +2102,141 @@ Example:
              error("Splash 1.8 or newer required")
          end
      end
+
+.. _splash-mouse-click:
+
+splash:mouse_click
+------------------
+
+Trigger mouse click event in web page.
+
+**Signature:** ``splash:mouse_click(x, y)``
+
+**Parameters:**
+
+* x - integer or float with x position of element to be clicked
+* y - integer or float with y position of element to be clicked
+
+**Returns:** nil
+
+**Async:**: no.
+
+
+.. note::
+
+    Coordinates for mouse events must be relative to web document. Element on which action is performed must be inside viewport (must be visible
+    to the user). If element is outside viewport and user needs to scroll to see it, you must either scroll
+    down with splash or simply set viewport to full with :ref:`splash-set-viewport-full`.
+
+.. note::
+
+    Mouse events are not propagated immediately, to see consequences of click  reflected in page source you must call :ref:`splash-wait`
+
+Example:
+
+.. code-block:: lua
+
+    -- get button element dimensions with javascript and perform mouse click
+
+    function main(splash)
+        assert(splash:go(splash.args.url))
+        get_dimensions = splash:jsfunc([[
+            function () {
+                rect = document.getElementById('button').getBoundingClientRect();
+                return {"x":rect.left, "y": rect.top}
+            }
+        ]])
+        dimensions = get_dimensions()
+        splash:mouse_click(dimensions.x, dimensions.y)
+        -- wait split second to allow event to propagate
+        splash:wait(0.1)
+        return splash:html()
+    end
+
+Example 2:
+
+.. code-block:: lua
+
+    -- click on element outside viewport
```

Comment:
this also works for clicking on element inside a viewport - what about keeping only a single example?